cli/alert.go

```diff
@@ -0,0 +1,180 @@
+package cli
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/prometheus/alertmanager/cli/format"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type alertmanagerAlertResponse struct {
+	Status    string       `json:"status"`
+	Data      model.Alerts `json:"data,omitempty"`
+	ErrorType string       `json:"errorType,omitempty"`
+	Error     string       `json:"error,omitempty"`
+}
+
+var alertFlags *flag.FlagSet
+
+// alertCmd represents the alert command
+var alertCmd = &cobra.Command{
+	Use:   "alert",
+	Short: "View and search through current alerts",
+	Long: `View and search through current alerts.
+
+  Amtool has a simplified prometheus query syntax, but contains robust support for
+  bash variable expansions. The non-option section of arguments constructs a list
+  of "Matcher Groups" that will be used to filter your query. The following
+  examples will attempt to show this behaviour in action:
+
+  amtool alert query alertname=foo node=bar
+
+  	This query will match all alerts with the alertname=foo and node=bar label
+  	value pairs set.
+
+  amtool alert query foo node=bar
+
+  	If alertname is ommited and the first argument does not contain a '=' or a
+  	'=~' then it will be assumed to be the value of the alertname pair.
+
+  amtool alert query 'alertname=~foo.*'
+
+  	As well as direct equality, regex matching is also supported. The '=~' syntax
+  	(similar to prometheus) is used to represent a regex match. Regex matching
+  	can be used in combination with a direct match.
+
+  amtool alert query alertname=foo node={bar,baz}
```

Comment:
Mh, how do you mean creating multiple silences? In the general case, there should be a single silence matching multiple labels if necessary rather than duplicated silences.

This comment is on alert querying, which is a different thing again.

I think I need a better example what's tried to be solved here.

In general, not growing representations for the same thing is preferable of course.